src/System.Memory/tests/Memory/Span.cs

```diff
@@ -106,5 +106,28 @@ public static void SpanFromDefaultMemory()
             Assert.True(spanObject.SequenceEqual(default));
         }
 
+        [Fact]
+        public static void TornMemory_Array_SpanThrowsIfOutOfBounds()
```

Comment:
Should we add similar tests for Pin as well?

Reply:
Yes, but I don't think such tests would currently pass. We need to fix the code as well.